### PR TITLE
Virt net mods batch 1

### PR DESCRIPTION
**Pull request by sxpert**
add a "modify" command that for now allows adding hosts (mac, name, ip) in the dhcp subsystem

### CODE REVIEW

**Comment by robynbergeron** [comment]
Thanks @sxpert. @drybjed please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by drybjed** [comment]
Looks good to me. shipit

**Comment by robynbergeron** [comment]
Thanks again to @sxpert for this PR, and thanks @drybjed for reviewing. Marking for inclusion.

**Comment by resmo** [comment]
@sxpert before we merge could you please add some words that modify has been added with version 2.1?

**Comment by sxpert** [comment]
@resmo that should do it ;-)

File: cloud/misc/virt_net.py [hunk 1]
```diff
@@ -206,6 +208,41 @@ def create(self, entryid):
...
+                mac_addr = new_data.get('mac')
+                hosts = old_data.xpath('/network/ip/dhcp/host')
+                # find the one mac we're looking for
+                host = None
+                for h in hosts :
+                    if h.get('mac') == mac_addr:
+                        host = h
+                        break
+                if host is None:
+                    # add the host
```
Comment (resmo) [inline cloud/misc/virt_net.py]: After review the code again, I think the check mode is not implemented the way it should.
IMHO it should be here and not on line 212, e.g.:

``` python
if self.module.check_mode:
    return True
```

File: cloud/misc/virt_net.py [hunk 2]
```diff
@@ -206,6 +208,41 @@ def create(self, entryid):
...
+                    res = network.update (libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+                        libvirt.VIR_NETWORK_SECTION_IP_DHCP_HOST,
+                        -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
+                    if res == 0: 
+                        return True
+                else:
+                    # change the host
+                    if host.get('name') == new_data.get('name') and host.get('ip') == new_data.get('ip'):
+                        return False
+                    else:
```
Comment (resmo) [inline cloud/misc/virt_net.py]: and here

``` python
if self.module.check_mode:
    return True
```

**Comment by resmo** [comment]
Could you clarify my question @sxpert @drybjed before we merge?

**Comment by sxpert** [comment]
I hadn't tried the check_mode, oops ;)
adding 

``` python
else:
  return True
```

at the end of the modify function should probably suffice

**Comment by resmo** [comment]
@sxpert not quite the way it is intended to be set. Correct me if I am wrong but now it always returns `changed=True` on `--check`, but it should only return `changed=True` if it would do any work without checkmode: 

``` python
...
                if host is None:
############# right before we make any serious work
                    if self.module.check_mode:
                        return True
#############
                    # add the host
                    res = network.update (libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
                        libvirt.VIR_NETWORK_SECTION_IP_DHCP_HOST,
                        -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
                    if res == 0: 
                        return True
...
```

**Comment by sxpert** [comment]
hmm, indeed, I'm rather tired these days ;-(

**Comment by gregdek** [comment]
@sxpert A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by sxpert** [comment]
@resmo does this version do what you think it should be doing ?

**Comment by resmo** [comment]
@sxpert looks good to me, thanks.

shipit

**Comment by gregdek** [comment]
@sxpert Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@sxpert A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@sxpert Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@sxpert A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by sxpert** [comment]
@gregdek can you look at it again ?

**Comment by resmo** [comment]
Hi @sxpert the failing travis is unrelated to your changes, I am merging in since you already received shipits.

**Comment by gregdek** [comment]
Thx @resmo!

**Pull request merged**